zapcore/json_encoder.go

```diff
@@ -302,7 +302,13 @@ func (enc *jsonEncoder) EncodeEntry(ent Entry, fields []Field) (*buffer.Buffer,
 	}
 	if ent.LoggerName != "" && final.NameKey != "" {
 		final.addKey(final.NameKey)
-		final.AppendString(ent.LoggerName)
+		cur := final.buf.Len()
+		final.EncodeLoggerName(ent.LoggerName, final)
```

Comment:
Same here - if the user-supplied logger name encoder is nil, we should fall back to the current behavior.